src/serverless/structure/serverless_parser.go

```diff
@@ -0,0 +1,317 @@
+	"io"
+	"math"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+)
+
+const FunctionTagsAttributeName = "tags"
+const StackTagsAttributeName = "stackTags"
+
+var TemplateSections = []string{"service", "provider", "tags", "stackTags", "resources", "functions", "region", "Resources"}
+
+type ServerlessTemplate struct {
+	Service  string `yaml:"service"`
+	Provider struct {
+		Name         string            `yaml:"name"`
+		Runtime      string            `yaml:"runtime"`
+		Region       string            `yaml:"region"`
+		ProviderTags map[string]string `yaml:"tags"`
+		CFNTags      map[string]string `yaml:"stackTags"`
+		Functions    interface{}       `yaml:"functions"`
+		Resources    struct {
+			Resources *cloudformation.Template `yaml:"Resources"`
+		} `yaml:"resources"`
+	} `yaml:"provider"`
+}
+
+type ServerlessParser struct {
+	rootDir              string
+	fileToResourcesLines map[string]common.Lines
+	template             *ServerlessTemplate
+}
+
+func (p *ServerlessParser) Init(rootDir string, _ map[string]string) {
+	p.rootDir = rootDir
+	p.fileToResourcesLines = make(map[string]common.Lines)
+	p.template = &ServerlessTemplate{}
+}
+
+func (p *ServerlessParser) GetSkippedDirs() []string {
+	return []string{}
+}
+
+func (p *ServerlessParser) GetSupportedFileExtensions() []string {
+	return []string{common.YamlFileType.Extension, common.YmlFileType.Extension}
+}
+
+func (p *ServerlessParser) ParseFile(filePath string) ([]structure.IBlock, error) {
+	parsedBlocks := make([]structure.IBlock, 0)
+	template, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("There was an error processing the serverless template: %s", err))
+	}
+	err = yaml.Unmarshal([]byte(template), p.template)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Unmarshal: %s", err), "SILENT")
+	}
+
+	if err != nil || template == nil {
+		logger.Error(fmt.Sprintf("There was an error processing the serverless template: %s", err), "SILENT")
+
+	}
+	//cfnStackTagsResource := p.template.Provider.CFNTags
+	functions := p.template.Provider.Functions
+	functionsMap := functions.(map[interface{}]interface{})
+	resourceNames := make([]string, 0)
+	var resourceNamesToLines map[string]*common.Lines
+	for funcName, _ := range functionsMap {
+		resourceNames = append(resourceNames, funcName.(string))
+	}
+	switch common.GetFileFormat(filePath) {
+	case common.YmlFileType.FileFormat, common.YamlFileType.FileFormat:
+		resourceNamesToLines = MapResourcesLineYAML(filePath, resourceNames)
+	default:
+		return nil, fmt.Errorf("unsupported file type %s", common.GetFileFormat(filePath))
+	}
+	minResourceLine := math.MaxInt8
+	maxResourceLine := 0
+	for _, funcName := range resourceNames {
+		var existingTags []tags.ITag
+		funcRange := functionsMap[funcName].(map[interface{}]interface{})
+		for key, val := range funcRange {
+			lines := resourceNamesToLines[funcName]
+			switch key {
+			case FunctionTagsAttributeName:
```

Comment:
@tronxd What happens in other cases? Are there going to be more cases in the future?

Reply:
No

Reply:
So I think it this switch statement can be replaced with a simple `if`